app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharecterView.kt

```diff
@@ -1,110 +1,71 @@
 package com.puzzlebench.clean_marvel_kotlin.presentation.mvp
 
-import android.app.Dialog
 import android.os.Bundle
-import android.support.v4.app.DialogFragment
-import android.support.v4.app.FragmentTransaction
-import android.support.v7.widget.GridLayoutManager
-import android.view.LayoutInflater
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
 import android.view.View
-import android.view.ViewGroup
 import com.puzzlebench.clean_marvel_kotlin.R
 import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
-import com.puzzlebench.clean_marvel_kotlin.presentation.extension.getImageByUrl
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
-import kotlinx.android.synthetic.main.activity_main.*
-import kotlinx.android.synthetic.main.character_dialog_fragment.*
-import kotlinx.android.synthetic.main.character_dialog_fragment.view.*
+import kotlinx.android.synthetic.main.activity_main.recycleView
+import kotlinx.android.synthetic.main.activity_main.progressBar
 import java.lang.ref.WeakReference
 
 
-class CharecterView(activity: MainActivity) {
+class CharacterView(activity: MainActivity) {
 
     companion object {
         const val EXTRA_CHARACTER = "character"
+        private const val TAG = "txn_tag"
     }
 
     private val activityRef = WeakReference(activity)
-    private val SPAN_COUNT = 1
-    var adapter = CharacterAdapter(emptyList(),
-            { character -> activity.applicationContext.showToast(character.name) },
-            { character -> activity.getCharacterById(character.id) })
+    var adapter = CharacterAdapter(emptyList()) { character ->
+                activity.applicationContext.showToast(character.name)
+                activity.getCharacterById(character.id)
+    }
 
     fun init() {
         val activity = activityRef.get()
-        if (activity != null) {
-            activity.recycleView.layoutManager = GridLayoutManager(activity, SPAN_COUNT)
-            activity.recycleView.adapter = adapter
+        activity?.let { act ->
```

Comment:
when you use let, try to use it and not create var like act. You can do this
activity?.let{
it.recyclerView.layout...
it.recyclerView.adapter
}